lightning/src/routing/gossip.rs

```diff
@@ -2077,9 +2077,9 @@ where
 				},
 				IndexedMapEntry::Vacant(node_entry) => {
 					let mut removed_node_counters = self.removed_node_counters.lock().unwrap();
-					**chan_info_node_counter = removed_node_counters
-						.pop()
-						.unwrap_or(self.next_node_counter.fetch_add(1, Ordering::Relaxed) as u32);
+					**chan_info_node_counter = removed_node_counters.pop().unwrap_or_else(|| {
```

Comment:
Could we add a call to `test_node_counter_consistency` to the end of this method? From local testing it looks like this would've caught the bug. Not sure if other methods could use it as well.

Reply:
FWIW, I get two test failures with this diff (i.e. reverting the fix and adding the check:
```diff
diff --git a/lightning/src/routing/gossip.rs b/lightning/src/routing/gossip.rs
index 9d85a6c58..f761e6a63 100644
--- a/lightning/src/routing/gossip.rs
+++ b/lightning/src/routing/gossip.rs
@@ -2077,9 +2077,9 @@ where
                                },
                                IndexedMapEntry::Vacant(node_entry) => {
                                        let mut removed_node_counters = self.removed_node_counters.lock().unwrap();
-                                       **chan_info_node_counter = removed_node_counters.pop().unwrap_or_else(|| {
+                                       **chan_info_node_counter = removed_node_counters.pop().unwrap_or(
                                                self.next_node_counter.fetch_add(1, Ordering::Relaxed) as u32
-                                       });
+                                       );
                                        node_entry.insert(NodeInfo {
                                                channels: vec![short_channel_id],
                                                announcement_info: None,
@@ -2088,6 +2088,9 @@ where
                                },
                        };
                }
+               core::mem::drop(channels);
+               core::mem::drop(nodes);
+               self.test_node_counter_consistency();

                Ok(())
        }